distribution/cli.go

```diff
@@ -0,0 +1,334 @@
+			BashComplete: corecommondocs.CreateBashCompletionFunc(),
+			Action: func(c *cli.Context) error {
+				return releaseBundleCreateCmd(c)
+			},
+		},
+		{
+			Name:         "release-bundle-update",
+			Flags:        cliutils.GetCommandFlags(cliutils.ReleaseBundleUpdate),
+			Aliases:      []string{"rbu"},
+			Description:  releasebundleupdate.Description,
+			HelpName:     corecommondocs.CreateUsage("ds rbu", releasebundleupdate.Description, releasebundleupdate.Usage),
+			UsageText:    releasebundleupdate.Arguments,
+			ArgsUsage:    common.CreateEnvVars(),
+			BashComplete: corecommondocs.CreateBashCompletionFunc(),
+			Action: func(c *cli.Context) error {
+				return releaseBundleUpdateCmd(c)
+			},
+		},
+		{
+			Name:         "release-bundle-sign",
+			Flags:        cliutils.GetCommandFlags(cliutils.ReleaseBundleSign),
+			Aliases:      []string{"rbs"},
+			Description:  releasebundlesign.Description,
+			HelpName:     corecommondocs.CreateUsage("ds rbs", releasebundlesign.Description, releasebundlesign.Usage),
+			UsageText:    releasebundlesign.Arguments,
+			ArgsUsage:    common.CreateEnvVars(),
+			BashComplete: corecommondocs.CreateBashCompletionFunc(),
+			Action: func(c *cli.Context) error {
+				return releaseBundleSignCmd(c)
+			},
+		},
+		{
+			Name:         "release-bundle-distribute",
+			Flags:        cliutils.GetCommandFlags(cliutils.ReleaseBundleDistribute),
+			Aliases:      []string{"rbd"},
+			Description:  releasebundledistribute.Description,
+			HelpName:     corecommondocs.CreateUsage("ds rbd", releasebundledistribute.Description, releasebundledistribute.Usage),
+			UsageText:    releasebundledistribute.Arguments,
+			ArgsUsage:    common.CreateEnvVars(),
+			BashComplete: corecommondocs.CreateBashCompletionFunc(),
+			Action: func(c *cli.Context) error {
+				return releaseBundleDistributeCmd(c)
+			},
+		},
+		{
+			Name:         "release-bundle-delete",
+			Flags:        cliutils.GetCommandFlags(cliutils.ReleaseBundleDelete),
+			Aliases:      []string{"rbdel"},
+			Description:  releasebundledelete.Description,
+			HelpName:     corecommondocs.CreateUsage("ds rbdel", releasebundledelete.Description, releasebundledelete.Usage),
+			UsageText:    releasebundledelete.Arguments,
+			ArgsUsage:    common.CreateEnvVars(),
+			BashComplete: corecommondocs.CreateBashCompletionFunc(),
+			Action: func(c *cli.Context) error {
+				return releaseBundleDeleteCmd(c)
+			},
+		},
+	})
+}
+
+func releaseBundleCreateCmd(c *cli.Context) error {
+	if !(c.NArg() == 2 && c.IsSet("spec") || (c.NArg() == 3 && !c.IsSet("spec"))) {
+		return cliutils.PrintHelpAndReturnError("Wrong number of arguments.", c)
+	}
+	if c.IsSet("detailed-summary") && !c.IsSet("sign") {
+		return cliutils.PrintHelpAndReturnError("The --detailed-summary option can't be used without --sign", c)
+	}
+	var releaseBundleCreateSpec *spec.SpecFiles
+	var err error
+	if c.IsSet("spec") {
+		releaseBundleCreateSpec, err = artifactory.GetSpec(c, true)
```

Comment:
Now that the ```GetSpec``` is used by both ```cli.go``` of Artifactory and Distribution, the right thing to do in terms of design is to move it out of ```artifactory/cli.go``` into ```cliutils/utils.go```